x-pack/plugins/maps/public/shared/layers/sources/es_geo_grid_source/convert_to_geojson.js

```diff
@@ -4,116 +4,102 @@
-
-      features = table.rows.map(row => {
-
-        const geohash = row[geohashColumn.id];
-        if (!geohash) return false;
-        const geohashLocation = decodeGeoHash(geohash);
-
-        let pointCoordinates;
-        if (geocentroidColumn) {
-          const location = row[geocentroidColumn.id];
-          pointCoordinates = [location.lon, location.lat];
-        } else {
-          pointCoordinates = [geohashLocation.longitude[2], geohashLocation.latitude[2]];
-        }
-
-        const rectangle = [
-          [geohashLocation.latitude[0], geohashLocation.longitude[0]],
-          [geohashLocation.latitude[0], geohashLocation.longitude[1]],
-          [geohashLocation.latitude[1], geohashLocation.longitude[1]],
-          [geohashLocation.latitude[1], geohashLocation.longitude[0]],
-        ];
-
-        const centerLatLng = [
-          geohashLocation.latitude[2],
-          geohashLocation.longitude[2]
-        ];
-
-        if (geoAgg.params.useGeocentroid) {
-          // see https://github.com/elastic/elasticsearch/issues/24694 for why clampGrid is used
-          pointCoordinates[0] = clampGrid(pointCoordinates[0], geohashLocation.longitude[0], geohashLocation.longitude[1]);
-          pointCoordinates[1] = clampGrid(pointCoordinates[1], geohashLocation.latitude[0], geohashLocation.latitude[1]);
-        }
-
-        const metrics = {};
-        metricColumns.forEach(metricColumn => {
-          metrics[metricColumn.aggConfig.id] = row[metricColumn.id];
-        });
-        //const value = row[metricColumn.id];
-        //min = Math.min(min, value);
-        //max = Math.max(max, value);
-
-        return {
-          type: 'Feature',
-          geometry: {
-            type: 'Point',
-            coordinates: pointCoordinates
-          },
-          properties: {
-            geohash: geohash,
-            geohash_meta: {
-              center: centerLatLng,
-              rectangle: rectangle
-            },
-            ...metrics
-          }
-        };
-
-
-      }).filter(row => row);
+const EMPTY_FEATURE_COLLECTION = {
+  type: 'FeatureCollection',
+  features: []
+};
 
-    }
+export function convertToGeoJson({ table, renderAs }) {
 
-  } else {
-    features = [];
+  if (!table || !table.rows) {
+    return EMPTY_FEATURE_COLLECTION;
   }
 
-  const featureCollection = {
-    type: 'FeatureCollection',
-    features: features
-  };
+  const geoGridColumn = table.columns.find(column => column.aggConfig.type.dslName === 'geotile_grid');
+
+  if (!geoGridColumn) {
+    return {
+      type: 'FeatureCollection',
+      features: []
+    };
+  }
+
+  const metricColumns = table.columns.filter(column => {
+    return column.aggConfig.type.type === 'metrics'
+      && column.aggConfig.type.dslName !== 'geo_centroid';
+  });
+  const geocentroidColumn = table.columns.find(column => column.aggConfig.type.dslName === 'geo_centroid');
```

Comment:
Check that `geocentroidColumn` and throw an error if does not exist since there is not longer a check before `geocentroidColumn.id`